Func/Results/ResultMapExtensionMethods.cs

```diff
@@ -0,0 +1,117 @@
+			valueTask.ContinueWith(result => result.Result.Then(value1 => func(value1).Map(Succeed)));
+
+		public static Task<Result<TResult>> ThenMap<TValue1, TResult>(this Result<TValue1> result, Func<TValue1, Task<TResult>> func) =>
+			result.Then(value1 => func(value1).Map(Succeed<TResult>));
+
+		public static Result<TResult> ThenMap<TValue1, TResult>(this Result<TValue1> result, Func<TValue1, TResult> func) =>
+			result.Then(value1 => func(value1).Map(Succeed));
+
+		public static Task<Result<TResult>> ThenMap<TValue1, TValue2, TResult>(this Task<Result<TValue2>> valueTask, Func<TValue1, TValue2, Task<TResult>> func, TValue1 value1) =>
+			valueTask.ContinueWith(result => result.Result.Then(value2 => func(value1, value2).Map(Succeed<TResult>))).Unwrap();
+
+		public static Task<Result<TResult>> ThenMap<TValue1, TValue2, TResult>(this Task<Result<TValue2>> valueTask, Func<TValue1, TValue2, TResult> func, TValue1 value1) =>
+			valueTask.ContinueWith(result => result.Result.Then(value2 => func(value1, value2).Map(Succeed)));
+
+		public static Task<Result<TResult>> ThenMap<TValue1, TValue2, TResult>(this Result<TValue2> result, Func<TValue1, TValue2, Task<TResult>> func, TValue1 value1) =>
+			result.Then(value2 => func(value1, value2).Map(Succeed<TResult>));
+
+		public static Result<TResult> ThenMap<TValue1, TValue2, TResult>(this Result<TValue2> result, Func<TValue1, TValue2, TResult> func, TValue1 value1) =>
+			result.Then(value2 => func(value1, value2).Map(Succeed));
+
+		public static Task<Result<TResult>> ThenMap<TValue1, TValue2, TValue3, TResult>(this Task<Result<TValue3>> valueTask, Func<TValue1, TValue2, TValue3, Task<TResult>> func, TValue1 value1, TValue2 value2) =>
+			valueTask.ContinueWith(result => result.Result.Then(value3 => func(value1, value2, value3).Map(Succeed<TResult>))).Unwrap();
+
+		public static Task<Result<TResult>> ThenMap<TValue1, TValue2, TValue3, TResult>(this Task<Result<TValue3>> valueTask, Func<TValue1, TValue2, TValue3, TResult> func, TValue1 value1, TValue2 value2) =>
+			valueTask.ContinueWith(result => result.Result.Then(value3 => func(value1, value2, value3).Map(Succeed)));
+
+		public static Task<Result<TResult>> ThenMap<TValue1, TValue2, TValue3, TResult>(this Result<TValue3> result, Func<TValue1, TValue2, TValue3, Task<TResult>> func, TValue1 value1, TValue2 value2) =>
+			result.Then(value3 => func(value1, value2, value3).Map(Succeed<TResult>));
+
+		public static Result<TResult> ThenMap<TValue1, TValue2, TValue3, TResult>(this Result<TValue3> result, Func<TValue1, TValue2, TValue3, TResult> func, TValue1 value1, TValue2 value2) =>
+			result.Then(value3 => func(value1, value2, value3).Map(Succeed));
+
+		public static Func<TValue2, TValue3, Result<TResult>> ThenMap<TValue1, TValue2, TValue3, TResult>(this Result<TValue1> result, Func<TValue1, TValue2, TValue3, TResult> func) =>
+			(value2, value3) => result.Then(value1 => func(value1, value2, value3).Map(Succeed));
```

Comment:
Just curious how this and similar methods to this is used. Looks like it takes a func with three parameters, which returns another func that takes two parameters?

Reply:
The general idea of them is that it's an easy way to create a partially applied function where you can provide the other values later.

However, I think there are a number of usability issues with them right now (type inference just doesn't work) and I don't think they're super useful so I've taken them out of this MR. Maybe they can be brought in later.

Reply:
Yeah I suspected so, was trying to use it in the code somewhere but couldnt seem to make it work at the time.